src/Mvc/Mvc.ApiExplorer/test/EndpointMetadataApiDescriptionProviderTest.cs

```diff
@@ -268,6 +270,20 @@ static void AssertPathParameter(ApiDescription apiDescription)
             AssertPathParameter(GetApiDescription(([FromRoute] int foo) => { }));
         }
 
+        [Fact]
+        public void AddsFromRouteParameterAsPathWithCustomType()
+        {
+            static void AssertPathParameter(ApiDescription apiDescription)
+            {
+                var param = Assert.Single(apiDescription.ParameterDescriptions);
+                Assert.Equal(typeof(TryParseStringRecord), param.Type);
+                Assert.Equal(typeof(TryParseStringRecord), param.ModelMetadata.ModelType);
```

Comment:
Oh this is why my PR was confusing to backport, we're touching the same file and you added this test in main already :D